File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        i,j,cnt,maxi=0,0,0,0
        for j in range(len(nums)):
            if nums[j]==0:
                cnt+=1
            while(cnt>k):
                if nums[i]==0:
                    cnt-=1
                i+=1
            maxi=max(maxi,j-i+1)
        return maxi

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int i=0;
        int cnt=0;
        int maxi=0;

        for(int j=0;j<nums.length;j++){
            if (nums[j]==0){
                cnt++;
            }
            while(cnt>k){
                if(nums[i]==0){
                    cnt--;
                }
                i++;
            }
            maxi=Math.max(maxi,j-i+1);

        }
        return maxi;
    }
}